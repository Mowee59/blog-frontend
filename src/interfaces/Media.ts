/*
 * This file is part of the Blog Frontend project.
 * 
 * Copyright (C) 2024 Aniss.dev
 * 
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 * 
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 * 
 * You should have received a copy of the GNU General Public License
 * along with this program.  If not, see <https://www.gnu.org/licenses/>.
 */


// Interface automatically generated by schemas-to-ts

import { MediaFormat } from './MediaFormat';
    export interface Media {
      id: number;
      attributes: {
        name: string;
        alternativeText: string;
        caption: string;
        width: number;
        height: number;
        formats: { thumbnail: MediaFormat; small: MediaFormat; medium: MediaFormat; large: MediaFormat; };
        hash: string;
        ext: string;
        mime: string;
        size: number;
        url: string;
        previewUrl: string;
        provider: string;
        createdAt: Date | string;
        updatedAt: Date | string;
      }
    }
    export interface Media_Plain {
      id: number;
      name: string;
      alternativeText: string;
      caption: string;
      width: number;
      height: number;
      formats: { thumbnail: MediaFormat; small: MediaFormat; medium: MediaFormat; large: MediaFormat; };
      hash: string;
      ext: string;
      mime: string;
      size: number;
      url: string;
      previewUrl: string;
      provider: string;
      createdAt: Date | string;
      updatedAt: Date | string;
    }
    


