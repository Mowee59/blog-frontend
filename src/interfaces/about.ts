import { Seo, Seo_Plain, Seo_NoRelations } from "./seo";

export interface About {
  id: number;
  attributes: {
    createdAt: Date;    updatedAt: Date;    publishedAt?: Date;    content?: string;
    seo?: Seo;
  };
}
export interface About_Plain {
  id: number;
  createdAt: Date;  updatedAt: Date;  publishedAt?: Date;  content?: string;
  seo?: Seo_Plain;
}

export interface About_NoRelations {
  id: number;
  createdAt: Date;  updatedAt: Date;  publishedAt?: Date;  content?: string;
  seo?: Seo_NoRelations;
}