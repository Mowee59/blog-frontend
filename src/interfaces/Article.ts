// Interface automatically generated by schemas-to-ts

import { Media } from "./Media";
import { Tag } from "./Tag";
import { Media_Plain } from "./Media";
import { Tag_Plain } from "./Tag";
import { Seo, Seo_NoRelations, Seo_Plain } from "./Seo";

export interface Article {
  id: number;
  attributes: {
    createdAt: Date | String;
    updatedAt: Date | String;
    publishedAt?: Date | String;
    title: string;
    coverImage: { data: Media };
    readingTime: string;
    content: string;
    slug: string;
    description: string;
    tags?: { data: Tag[] };
    seo?: Seo;
    locale: string;
    localizations?: { data: Article[] };
  };
}
export interface Article_Plain {
  id: number;
  createdAt: Date | string;
  updatedAt: Date | string;
  publishedAt?: Date | string;
  title: string;
  coverImage: Media_Plain;
  readingTime: string;
  content: string;
  slug: string;
  description: string;
  tags?: Tag_Plain[];
  seo?: Seo_Plain;
  locale: string;
  localizations?: Article_Plain[];
}

export interface Article_NoRelations {
  id: number;
  createdAt: Date | string;
  updatedAt: Date | string;
  publishedAt?: Date | String;
  title: string;
  coverImage: number;
  readingTime: string;
  content: string;
  slug: string;
  description: string;
  tags?: number[];
  seo?: Seo_NoRelations;
  locale: string;
  localizations?: Article[];
}

export interface Article_Preview {
  id: number;
  attributes: {
    publishedAt: string;
    title: string;
    coverImage: { data: Media };
    slug: string;
    description: string;
    tags: { data: Tag[] };
  };
}
