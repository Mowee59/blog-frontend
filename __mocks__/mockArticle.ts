import { Article_Preview } from "@/interfaces/article";
import { Media } from "@/interfaces/Media";

export const mockArticle: Article_Preview = {
	id: 2,
	attributes: {
		title: "Article test",
		slug: "article-test",
	
		publishedAt: "2024-09-10T16:16:35.724Z",
		description: "This is a test article summary",
		coverImage: {
			data: {
				id: 5,
				attributes: {
					name: "coverARticleTest.webp",
					alternativeText: "CoverImage",
					caption: "La description de l'image",
					width: 6000,
					height: 4000,
					formats: {
						thumbnail: {
							name: "thumbnail_coverARticleTest.webp",
							hash: "thumbnail_cover_A_Rticle_Test_44ad5169b1",
							ext: ".webp",
							mime: "image/webp",
							path: '',
							width: 234,
							height: 156,
							size: 4.9,
							url: "https://example.com/image.jpg"
						},
						small: {
							name: "small_coverARticleTest.webp",
							hash: "small_cover_A_Rticle_Test_44ad5169b1",
							ext: ".webp",
							mime: "image/webp",
							path: '',
							width: 500,
							height: 333,
							size: 13.1,
							url: "https://example.com/image1.jpg"
						},
						large: {
							name: "large_coverARticleTest.webp",
							hash: "large_cover_A_Rticle_Test_44ad5169b1",
							ext: ".webp",
							mime: "image/webp",
							path: '',
							width: 1000,
							height: 667,
							size: 33.6,
							url: "https://example.com/image2.jpg"
						},
						medium: {
							name: "medium_coverARticleTest.webp",
							hash: "medium_cover_A_Rticle_Test_44ad5169b1",
							ext: ".webp",
							mime: "image/webp",
							path: '',
							width: 750,
							height: 500,
							size: 22.59,
							url: "https://example.com/image3.jpg"
						}
					},
					hash: "cover_A_Rticle_Test_44ad5169b1",
					ext: ".webp",
					mime: "image/webp",
					size: 1282.64,
					url: "https://example.com/imageOriginal.jpg",
					previewUrl: '',
					provider: "local",
					createdAt: "2024-09-18T18:00:47.473Z",
					updatedAt: "2024-09-25T17:35:19.841Z"
				}
			}
		},
		tags: {
			data: [
				{
					id: 1,
					attributes: {
						cover: { data: {} as Media },
						name: "Algorithmie",
						createdAt: "2024-09-11T08:17:24.849Z",
						updatedAt: "2024-09-27T10:18:29.586Z",
						locale: "fr",
						description: "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Fusce mattis id augue non feugiat. Quisque rhoncus diam ornare ex feugiat, ac commodo odio aliquam. Vestibulum condimentum auctor tortor, ut laoreet lorem finibus et. Maecenas vel pulvinar eros, in porttitor metus. Donec tempor quis diam vitae interdum. Quisque mollis finibus purus, ac malesuada lectus auctor ut. In nisi tellus, ultricies non aliquam nec, sodales id turpis."
					}
				},
				{
					id: 3,
					attributes: {
						cover: { data: {} as Media },
						name: "Javascript",
						createdAt: "2024-09-19T13:58:51.645Z",
						updatedAt: "2024-09-27T10:18:36.955Z",
						locale: "fr",
						description: "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Fusce mattis id augue non feugiat. Quisque rhoncus diam ornare ex feugiat, ac commodo odio aliquam. Vestibulum condimentum auctor tortor, ut laoreet lorem finibus et. Maecenas vel pulvinar eros, in porttitor metus. Donec tempor quis diam vitae interdum. Quisque mollis finibus purus, ac malesuada lectus auctor ut. In nisi tellus, ultricies non aliquam nec, sodales id turpis."
					}
				},
				{
					id: 4,
					attributes: {
						cover: { data: {} as Media },
						name: "React",
						createdAt: "2024-09-20T10:30:18.922Z",
						updatedAt: "2024-09-27T10:18:40.751Z",
						locale: "fr",
						description: "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Fusce mattis id augue non feugiat. Quisque rhoncus diam ornare ex feugiat, ac commodo odio aliquam. Vestibulum condimentum auctor tortor, ut laoreet lorem finibus et. Maecenas vel pulvinar eros, in porttitor metus. Donec tempor quis diam vitae interdum. Quisque mollis finibus purus, ac malesuada lectus auctor ut. In nisi tellus, ultricies non aliquam nec, sodales id turpis."
					}
				}
			]
		},
	
	}}