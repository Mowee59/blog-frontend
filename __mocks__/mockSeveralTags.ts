// New file: __mocks__/mockSeveralTags.ts
import { Article, Article_NoRelations } from '@/interfaces/Article';
import { Tag } from '@/interfaces/Tag';

export const mockSeveralTags: Tag[] = [
  {
    id: 1,
    attributes: {
      name: "React",
      locale: "fr",
      createdAt: "2024-09-11T08:17:24.849Z",
      updatedAt: "2024-09-27T10:18:29.586Z",
      description: "Une librairie Javascript pour construire des interfaces utilisateur",
      cover: {
        data: {
          id: 3,
          attributes: {
            name: "react.webp",
            alternativeText: "Image de couverture catégorie React",
            caption: "React",
            width: 6000,
            height: 4000,
            formats: {
              thumbnail: {
                name: "thumbnail_react.webp",
                hash: "thumbnail_react_8111d1aa1b",
                ext: ".webp",
                mime: "image/webp",
                path: '',
                width: 234,
                height: 156,
                size: 3.86,
                url: "/uploads/thumbnail_react_8111d1aa1b.webp"
              },
              small: {
                name: "small_react.webp",
                hash: "small_react_8111d1aa1b",
                ext: ".webp",
                mime: "image/webp",
                path: '',
                width: 300,
                height: 200,
                size: 5.21,
                url: "/uploads/small_react_8111d1aa1b.webp"
              },
              medium: {
                name: "medium_react.webp",
                hash: "medium_react_8111d1aa1b",
                ext: ".webp",
                mime: "image/webp",
                path: '',
                width: 750,
                height: 500,
                size: 15.35,
                url: "/uploads/medium_react_8111d1aa1b.webp"
              },
              large: {
                name: "large_react.webp",
                hash: "large_react_8111d1aa1b",
                ext: ".webp",
                mime: "image/webp",
                path: '',
                width: 1000,
                height: 667,
                size: 21.93,
                url: "/uploads/large_react_8111d1aa1b.webp"
              }
            },
            hash: "react_8111d1aa1b",
            ext: ".webp",
            mime: "image/webp",
            size: 3633.27,
            url: "/uploads/react_8111d1aa1b.webp",
            previewUrl: '',
            provider: "local",
            createdAt: "2024-09-11T08:17:11.574Z",
            updatedAt: "2024-09-20T09:36:16.491Z"
          }
        }
      },
      articles: {
        data: [{} as Article, {} as Article, {} as Article]
      },
      localizations: {
        data: []
      }
    }
  },
  {
    id: 2,
    attributes: {
      name: "TypeScript",
      locale: "fr",
      createdAt: "2024-09-12T10:20:30.123Z",
      updatedAt: "2024-09-28T11:22:33.456Z",
      description: "Un sur-ensemble typé de JavaScript qui se compile en JavaScript pur",
      cover: {
        data: {
          id: 4,
          attributes: {
            name: "typescript.webp",
            alternativeText: "Image de couverture catégorie TypeScript",
            caption: "TypeScript",
            width: 6000,
            height: 4000,
            formats: {
              thumbnail: {
                name: "thumbnail_typescript.webp",
                hash: "thumbnail_typescript_9222e2bb2c",
                ext: ".webp",
                mime: "image/webp",
                path: '',
                width: 234,
                height: 156,
                size: 4.12,
                url: "/uploads/thumbnail_typescript_9222e2bb2c.webp"
              },
              small: {
                name: "small_typescript.webp",
                hash: "small_typescript_9222e2bb2c",
                ext: ".webp",
                mime: "image/webp",
                path: '',
                width: 300,
                height: 200,
                size: 5.67,
                url: "/uploads/small_typescript_9222e2bb2c.webp"
              },
              medium: {
                name: "medium_typescript.webp",
                hash: "medium_typescript_9222e2bb2c",
                ext: ".webp",
                mime: "image/webp",
                path: '',
                width: 750,
                height: 500,
                size: 16.89,
                url: "/uploads/medium_typescript_9222e2bb2c.webp"
              },
              large: {
                name: "large_typescript.webp",
                hash: "large_typescript_9222e2bb2c",
                ext: ".webp",
                mime: "image/webp",
                path: '',
                width: 1000,
                height: 667,
                size: 24.56,
                url: "/uploads/large_typescript_9222e2bb2c.webp"
              }
            },
            hash: "typescript_9222e2bb2c",
            ext: ".webp",
            mime: "image/webp",
            size: 3987.65,
            url: "/uploads/typescript_9222e2bb2c.webp",
            previewUrl: '',
            provider: "local",
            createdAt: "2024-09-12T10:20:15.789Z",
            updatedAt: "2024-09-21T12:34:56.789Z"
          }
        }
      },
      articles: {
        data: [{} as Article, {} as Article]
      },
      localizations: {
        data: []
      }
    }
  },
  {
    id: 3,
    attributes: {
      name: "Next.js",
      locale: "fr",
      createdAt: "2024-09-13T14:25:36.789Z",
      updatedAt: "2024-09-29T15:27:39.012Z",
      description: "Un framework React pour la production",
      cover: {
        data: {
          id: 5,
          attributes: {
            name: "nextjs.webp",
            alternativeText: "Image de couverture catégorie Next.js",
            caption: "Next.js",
            width: 6000,
            height: 4000,
            formats: {
              thumbnail: {
                name: "thumbnail_nextjs.webp",
                hash: "thumbnail_nextjs_3333f3cc3d",
                ext: ".webp",
                mime: "image/webp",
                path: '',
                width: 234,
                height: 156,
                size: 3.98,
                url: "/uploads/thumbnail_nextjs_3333f3cc3d.webp"
              },
              small: {
                name: "small_nextjs.webp",
                hash: "small_nextjs_3333f3cc3d",
                ext: ".webp",
                mime: "image/webp",
                path: '',
                width: 300,
                height: 200,
                size: 5.43,
                url: "/uploads/small_nextjs_3333f3cc3d.webp"
              },
              medium: {
                name: "medium_nextjs.webp",
                hash: "medium_nextjs_3333f3cc3d",
                ext: ".webp",
                mime: "image/webp",
                path: '',
                width: 750,
                height: 500,
                size: 15.87,
                url: "/uploads/medium_nextjs_3333f3cc3d.webp"
              },
              large: {
                name: "large_nextjs.webp",
                hash: "large_nextjs_3333f3cc3d",
                ext: ".webp",
                mime: "image/webp",
                path: '',
                width: 1000,
                height: 667,
                size: 22.76,
                url: "/uploads/large_nextjs_3333f3cc3d.webp"
              }
            },
            hash: "nextjs_3333f3cc3d",
            ext: ".webp",
            mime: "image/webp",
            size: 3845.92,
            url: "/uploads/nextjs_3333f3cc3d.webp",
            previewUrl: '',
            provider: "local",
            createdAt: "2024-09-13T14:25:30.123Z",
            updatedAt: "2024-09-22T16:38:59.012Z"
          }
        }
      },
      articles: {
        data: [{} as Article, {} as Article, {} as Article, {} as Article]
      },
      localizations: {
        data: []
      }
    }
  }
];