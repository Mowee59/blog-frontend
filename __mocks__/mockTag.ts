import { Article, Article_NoRelations } from '@/interfaces/Article';
import { Tag } from '@/interfaces/Tag';

export const mockTag: Tag = {
  id: 1,
  attributes: {
    name: "React",
    locale: "fr",
    createdAt: "2024-09-11T08:17:24.849Z",
    updatedAt: "2024-09-27T10:18:29.586Z",
    description: "Une librairie Javascript pour construire des interfaces utilisateur",
    cover: {
      data: {
        id: 3,
        attributes: {
          name: "algo.webp",
          alternativeText: "Image de couverture cat√©gorie Algorithmie",
          caption: "Alogrithmie",
          width: 6000,
          height: 4000,
          formats: {
            thumbnail: {
              name: "thumbnail_algo.webp",
              hash: "thumbnail_algo_8111d1aa1b",
              ext: ".webp",
              mime: "image/webp",
              path: '',
              width: 234,
              height: 156,
              size: 3.86,
              url: "/uploads/thumbnail_algo_8111d1aa1b.webp"
            },
            small: {
              name: "small_algo.webp",
              hash: "small_algo_8111d1aa1b",
              ext: ".webp",
              mime: "image/webp",
              path: '',
              width: 300,
              height: 200,
              size: 5.21,
              url: "/uploads/small_algo_8111d1aa1b.webp"
            },
            medium: {
              name: "medium_algo.webp",
              hash: "medium_algo_8111d1aa1b",
              ext: ".webp",
              mime: "image/webp",
              path: '',
              width: 750,
              height: 500,
              size: 15.35,
              url: "/uploads/medium_algo_8111d1aa1b.webp"
            },
            large: {
              name: "large_algo.webp",
              hash: "large_algo_8111d1aa1b",
              ext: ".webp",
              mime: "image/webp",
              path: '',
              width: 1000,
              height: 667,
              size: 21.93,
              url: "/uploads/large_algo_8111d1aa1b.webp"
            }
          },
          hash: "algo_8111d1aa1b",
          ext: ".webp",
          mime: "image/webp",
          size: 3633.27,
          url: "/uploads/algo_8111d1aa1b.webp",
          previewUrl: '',
          provider: "local",
          createdAt: "2024-09-11T08:17:11.574Z",
          updatedAt: "2024-09-20T09:36:16.491Z"
        }
      }
    },
    articles: {
      data: [  {} as Article, {} as Article , {} as Article]
    },
    localizations: {
      data: []
    }
  }
}

