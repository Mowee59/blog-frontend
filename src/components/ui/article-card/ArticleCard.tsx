import React from "react";
import Image from "next/image";
import { Article, Article_Preview } from "@/interfaces/article";

type ArticleCardProps = {
  article: Article_Preview;
};

const ArticleCard = (props: ArticleCardProps) => {
  // Asssigning the thumbnail to a varialbe
  const coverImageThumbnail =
    props.article.attributes.coverImage.data.attributes.formats.thumbnail;

  // Creating a date from publciated at attribute
  const publicationDate = new Date(props.article.attributes.publishedAt!);

  // The array of tags the article is related with
  const tags = props.article.attributes.tags!.data;

  return (
    <article className=" cursor-pointer @container">
      <div className="flex h-fit  w-full min-w-full justify-between gap-5 @lg:gap-8 @3xl:gap-12 md:h-[120px]">
        <div className="">
          <h3 className=" mb-3 text-xs font-medium leading-3 text-neutral-600 @lg:mb-2 dark:text-neutral-400">
            {`${publicationDate?.getDate()} `}
            <span className="text-transform: capitalize">
              {`${publicationDate?.toLocaleDateString("FR-fr", { month: "short" }).slice(0, -2)} `}
              &#8226;{" "}
            </span>
            <div className="inline-block ">
              {tags.map((tag) => (
                <span key={tag.id} className="mr-1 hover:underline">
                  {"#" + tag.attributes.name + " "}
                </span>
              ))}
            </div>
          </h3>
          <h2 className="  max-h-full text-xl font-semibold  leading-tight text-neutral-800 dark:text-neutral-300 sm:text-base sm:leading-normal lg:text-lg lg:leading-relaxed">
            {props.article.attributes.title}
          </h2>
          <p className="line-clamp-2  max-h-full text-sm font-medium  leading-tight text-neutral-800 dark:text-neutral-300 sm:text-base sm:leading-normal lg:text-lg lg:leading-relaxed">
            {props.article.attributes.description}
          </p>
        </div>
        <div className="relative  hidden h-[120px] w-[160px] flex-shrink-0 @md:block">
          <Image
            src={coverImageThumbnail.url}
            alt={
              props.article.attributes.coverImage.data.attributes
                .alternativeText
            }
            fill
            className=" rounded "
          />
        </div>
      </div>
    </article>
  );
};

export default ArticleCard;
