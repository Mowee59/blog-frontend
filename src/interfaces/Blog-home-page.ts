/*
 * This file is part of the Blog Frontend project.
 * 
 * Copyright (C) 2024 Aniss.dev
 * 
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 * 
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 * 
 * You should have received a copy of the GNU General Public License
 * along with this program.  If not, see <https://www.gnu.org/licenses/>.
 */


// Interface automatically generated by schemas-to-ts

import { Article } from "./Article";
import { Article_Plain } from "./Article";
import { Seo, Seo_NoRelations, Seo_Plain } from "./Seo";
export interface BlogHomePage {
  id: number;
  attributes: {
    createdAt: Date;
    updatedAt: Date;
    publishedAt?: Date;
    seo?: Seo;
    featuredArticle?: { data: Article };
    locale: string;
    localizations?: { data: BlogHomePage[] };
  };
}
export interface BlogHomePage_Plain {
  id: number;
  createdAt: Date;
  updatedAt: Date;
  publishedAt?: Date;
  seo?: Seo_Plain
  featuredArticle?: Article_Plain;
  locale: string;
  localizations?: BlogHomePage_Plain[];
}

export interface BlogHomePage_NoRelations {
  id: number;
  createdAt: Date;
  updatedAt: Date;
  publishedAt?: Date;
  featuredArticle?: number;
  seo?: Seo_NoRelations;
  locale: string;
  localizations?: BlogHomePage[];
}
