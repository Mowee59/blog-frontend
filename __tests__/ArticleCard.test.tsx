import React from 'react';
import { render, screen } from '@testing-library/react';
import ArticleCard from "../src/components/ui/article-card/ArticleCard";
import { Article_Preview } from '@/interfaces/Article';
import { Media } from '@/interfaces/Media';
import { mockArticle } from '../__mocks__/mockArticle';

describe('ArticleCard', () => {


  it('renders the article title', () => {
    render(<ArticleCard article={mockArticle} />);
    expect(screen.getByText('Article test')).toBeInTheDocument();
  });

	it('displays the article tags', () => {
    render(<ArticleCard article={mockArticle} />);
    
    mockArticle.attributes.tags.data.forEach(tag => {
      expect(screen.getByText(`#${tag.attributes.name}`)).toBeInTheDocument();
    });
  });
  

  it('shows the article date', () => {
    // Render the ArticleCard component with the mock article
    render(<ArticleCard article={mockArticle} />);

    // Use a custom function to find the date element
    const dateElement = screen.getByText((content, element) => {
      // Ensure the element is an h3 tag
      if (element?.tagName.toLowerCase() !== 'h3') return false;

      // Check if the content includes the day (10)
      const hasDay = content.includes('10');

      // Check if the element's text content includes the month (sep)
      const hasMonth = element.textContent?.toLowerCase().includes('sep');

      // Return true if both day and month are present or false if not so we are sure to return a boolean
      return (hasDay && hasMonth) || false;
    });

    // Assert that the date element is in the document
    expect(dateElement).toBeInTheDocument();
  });

  it('renders the article summary', () => {
    render(<ArticleCard article={mockArticle} />);
    expect(screen.getByText('This is a test article summary')).toBeInTheDocument();
  });

  it('displays the article image', () => {
    render(<ArticleCard article={mockArticle} />);
    const image = screen.getByRole('img');
    expect(image).toHaveAttribute('src', 'https://example.com/image.jpg');
    expect(image).toHaveAttribute('alt', 'CoverImage');
  });

  it('matches snapshot', () => {
    const { asFragment } = render(<ArticleCard article={mockArticle} />);
    expect(asFragment()).toMatchSnapshot();
  });

  // Add more tests as needed
});