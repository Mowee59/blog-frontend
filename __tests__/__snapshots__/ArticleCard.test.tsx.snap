// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticleCard matches snapshot 1`] = `
<DocumentFragment>
  <a
    href="/articles/article-test"
  >
    <article
      class=" cursor-pointer @container"
    >
      <div
        class="flex h-fit  w-full min-w-full justify-between gap-5 @lg:gap-8 @3xl:gap-12 md:h-[120px]"
      >
        <div
          class=""
        >
          <h3
            class=" mb-3 text-xs font-medium leading-3 text-neutral-600 @lg:mb-2 dark:text-neutral-400"
          >
            10 
            <span
              class="text-transform: capitalize"
            >
              sep â€¢ 
            </span>
            <div
              class="inline-block "
            >
              <span
                class="mr-1 hover:underline"
              >
                #Algorithmie 
              </span>
              <span
                class="mr-1 hover:underline"
              >
                #Javascript 
              </span>
              <span
                class="mr-1 hover:underline"
              >
                #React 
              </span>
            </div>
          </h3>
          <h2
            class="  max-h-full text-xl font-semibold  leading-tight text-neutral-800 dark:text-neutral-300 sm:text-base sm:leading-normal lg:text-lg lg:leading-relaxed"
          >
            Article test
          </h2>
          <p
            class="line-clamp-2  max-h-full text-sm font-medium  leading-tight text-neutral-800 dark:text-neutral-300 sm:text-base sm:leading-normal lg:text-lg lg:leading-relaxed"
          >
            This is a test article summary
          </p>
        </div>
        <div
          class="relative  hidden h-[120px] w-[160px] flex-shrink-0 @md:block"
        >
          <img
            alt="CoverImage"
            class=" rounded "
            src="https://example.com/image.jpg"
          />
        </div>
      </div>
    </article>
  </a>
</DocumentFragment>
`;
